office-for-students/beta-data-pipelines
# Feature/build course search

### What

Build new function app to build course search index and populate it with course data.

### How to review

1) Check code
2) Check unit tests pass
3) Add the following config variables to your local.settings.json file:
```
    "CourseSearchBuilerContainerName": "course-search-blobs",
    "SearchURL": {searchURL},
    "SearchAPIKey": {searchAPIKey},
    "AzureSearchAPIVersion": "2019-05-06",
```
4) You can trigger the process by dropping a new blob into `course-search-blobs` storage container
5) See the courses being built (takes between 2 and 5 minutes depending on internet connection speed)
6) Go to azure search in portal, select instance and check the total number of courses matches cosmos db; it may take up to 5minutes after function has completed for the numbers to sync
